### PR TITLE
[MB-3508] React-Query boilerplate

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR adds initial boilerplate code for using React-Query to make API calls in the Office app.
- Install react-query & react-query-devtools
- Add global config for ReactQuery and dev tools to the Office app
- Use ReactQuery for all API calls on PaymentRequestIndex & PaymentRequestReview:
  - query `paymentRequests`
  - query `paymentRequests` by ID, and `mtoShipments, mtoServiceItems` by MTO ID
  - `mutatePaymentRequest` for updating Payment Request status
  - `mutatePaymentServiceItemStatus` for updating Payment Service Item status
- Write custom hook for loading a single payment request, and its related MTO shipments & MTO service items (this was done primarily to serve as an example of chaining API queries)
- Adds some utils for dealing with API data (checking statuses of multiple queries, mapping object to array)
- Moves PaymentRequestIndex & PaymentRequestReview into `src/pages/Office` directories
  - This change makes it hard to see the actual code diff, check out [this commit range](https://github.com/transcom/mymove/pull/4529/files/cbee8fddab8c7e468bd470096285d9087238a5ef) for changes prior to the files moving
- Add tests for new code
  - I added two different examples of testing a component that uses React Query:
  - the `PaymentRequestIndex` test creates a mock cache with pre-loaded data to demonstrate the component loading with data from a single query that has already been cached
  - the `PaymentRequestReview` test mocks the response to the custom hook which is making multiple queries behind the scenes

This PR leaves a few things open for us to decide down the road:
- Do we want to implement any automatic retry logic?
- How/where to log query/mutation errors
- Any improvements/clean up for the new API code

## Reviewer Notes

- In local dev you will see the React Query flower in the bottom left to open/close devtools. This should not appear in production builds, or in any deployed environments.
- Main thing to verify is that you are still able to log in as a TIO and complete the Payment Request review flow. No user-facing changes or deprecations are expected.

## Setup

- Run the Office app and log in as a TIO user
- Verify you can view the Payment Requests index
- Click on a Payment Request and verify you can approve & reject service items
- After approving/rejecting all service items, verify you can complete review of the payment request

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/backend#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/backend#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3508) for this change

### CODE REVIEW

File: src/hooks/queries.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable import/prefer-default-export */
+import { useQuery } from 'react-query';
+
+import { getPaymentRequest, getMTOShipments, getMTOServiceItems } from 'services/ghcApi';
+import { getQueriesStatus } from 'utils/api';
+
+export const usePaymentRequestQueries = (paymentRequestId) => {
+  // get payment request by ID
+  const { data: { paymentRequests, paymentServiceItems } = {}, ...paymentRequestQuery } = useQuery(
+    ['paymentRequests', paymentRequestId],
```
Comment (duncan-truss) [inline src/hooks/queries.js:10]: We could definitely turn the various useQuery calls into functions so we don't end up using different cache keys all over the place:

```
function usePaymentRequestQuery(paymentRequestId, options) {
    return useQuery(['paymentRequests', paymentRequestId], getPaymentRequest, options)
}
```

File: src/utils/api.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// eslint-disable-next-line security/detect-object-injection
+export const mapObjectToArray = (obj) => Object.keys(obj).map((i) => obj[i]);
+
+export const getQueriesStatus = (queries) => {
+  // Queries should be an array of statuses returned by useQuery (https://react-query.tanstack.com/docs/api#usequery)
+  return {
+    isLoading: queries.some((q) => q.status === 'loading'),
+    isError: queries.some((q) => q.status === 'error'),
+    isSuccess: queries.every((q) => q.status === 'success'),
+    errors: queries.reduce((errors, q) => (q.error ? [...errors, q.error] : errors), []),
```
Comment (duncan-truss) [inline src/utils/api.js:13]: Do we care about the idle statuses here? If we get a result from isLoading should we continue looking up the other statuses? Similarly if isError is false maybe we can forgo building the error list?

File: src/utils/api.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/prefer-default-export */
+// utility functions related to API interactions
+
+// eslint-disable-next-line security/detect-object-injection
+export const mapObjectToArray = (obj) => Object.keys(obj).map((i) => obj[i]);
```
Comment (duncan-truss) [inline src/utils/api.js:5]: Is this the same as `Object.values(obj)`?

Comment (suzubara) [reply src/utils/api.js:5]: oh yes 🙃

File: src/hooks/queries.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  // get MTO service items
+  const { data: { mtoServiceItems } = {}, ...mtoServiceItemQuery } = useQuery(
+    ['mtoServiceItems', mtoID],
+    getMTOServiceItems,
+    {
+      enabled: !!mtoID,
+    },
+  );
+
+  const { isLoading, isError, isSuccess } = getQueriesStatus([
```
[the remark this answers is not in the document]
Comment (suzubara) [reply src/hooks/queries.js:31]: yeah... an error coming back from a query probably means there is either a bug in the code, or the user is not authenticated and that will most likely be handled globally.

**Comment by suzubara** [comment]
I believe the integration test that's failing here is due to the GHC API not being turned on in CI tests, and will be fixed when https://github.com/transcom/mymove/pull/4565 is merged in

File: src/pages/Office/PaymentRequestReview/PaymentRequestReview.jsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      const newPaymentServiceItem = data.paymentServiceItems[variables.paymentServiceItemID];
+      queryCache.setQueryData([PAYMENT_REQUESTS, paymentRequestId], {
+        paymentRequests,
+        paymentServiceItems: {
+          ...paymentServiceItems,
+          [`${variables.paymentServiceItemID}`]: newPaymentServiceItem,
+        },
+      });
+    },
+  });
```
Comment (donaldthai) [inline src/pages/Office/PaymentRequestReview/PaymentRequestReview.jsx:59]: **question:** Should we move these mutation functions into its own file too? Like the queries?

File: src/App/index.jsx [hunk 6]
```diff
@@ -39,18 +41,32 @@ const officeContext = { ...defaultOfficeContext, flags };
 const myMoveContext = { ...defaultMyMoveContext, flags };
 const adminContext = { ...defaultAdminContext, flags };
 
+const officeQueryConfig = {
+  queries: {
+    retry: false, // default to no retries for now
+    refetchOnWindowFocus: true,
+    // onError: noop, // TODO - log errors?
```
Comment (gidjin) [inline src/App/index.jsx:48]: logging errors seems valuable. If things are not working right end users can at the very least be asked to look for errors in the console. Probably need some way for bubbling things like this up, but certainly, that's not part of this PR and likely on your radar already.

File: src/pages/Office/PaymentRequestReview/PaymentRequestReview.jsx (hunk 5 above)
Comment (suzubara) [reply src/pages/Office/PaymentRequestReview/PaymentRequestReview.jsx:59]: it may be a good thing to do eventually, especially if any mutations can be called from multiple places. in the interest of not over-architecting this example though I will leave it as-is.

File: src/App/index.jsx (hunk 6 above)
Comment (suzubara) [reply src/App/index.jsx:48]: yeah, since we have lint rules against printing to the console (and I think @tinyels said something about not exposing our stack, although that is not very applicable to the F/E code) I left this as an outstanding question. If we end up integrating something like Sentry then this can be where we send error logs.
